### DISCUSSION TITLE
#Help I can't run conda create -n f5-tts python=3.1x

### DISCUSSION THREAD

**Discussion by SyamsQ** [SWivid/F5-TTS#541] [Q&A]
Why I stuck here? How to fix this?
![image](https://github.com/user-attachments/assets/748a33f0-0dd7-46f6-8c77-615910313acd)
![image](https://github.com/user-attachments/assets/aa3ff1ca-2e21-4bb9-9f5d-948ea3772dd6)

**Comment by SWivid** [comment]
need a conda first
choose from 
https://www.anaconda.com/download
or
https://docs.anaconda.com/miniconda/

and you may also need to install ffmpeg #162 #297

may also check for online blogs or video tutorial, don't worry, it's easy

**Reply by SyamsQ** [reply]
Isn't my PC already installed with Ccnda?

**Reply by SyamsQ** [reply]
I tried to follow tutorials on Youtube. Of course, I wondered why it was stuck at Conda. Even though conda is already installed on my PC. ![image](https://github.com/user-attachments/assets/47393160-a0fb-41e2-973e-3787cadacb36)
![image](https://github.com/user-attachments/assets/f0d7cdbc-6cc6-45e3-b676-df0617c7f887)

**Reply by SWivid** [reply]
maybe during installation you haven't click on something like "add to path"
this is to ensure you can call conda directly in command line

you can do the installation again and be careful to click on this option, or to manually add conda to system PATH

**Reply by SyamsQ** [reply]
Please help. Why do I still fail even though I installed Conda?
![image](https://github.com/user-attachments/assets/1b9f05bb-7cc1-4daa-8793-890670471c1e)
![image](https://github.com/user-attachments/assets/abd33152-e99d-44a5-af4b-6dc112f3e3d4)
![image](https://github.com/user-attachments/assets/22680fbd-47a7-43a3-a8ab-fdda4739cd0e)

**Reply by SWivid** [reply]
after the progress bar finish, are there any thing like "Add to PATH"?

**Reply by SyamsQ** [reply]
Which progress are you talking about? ![image](https://github.com/user-attachments/assets/e19538ef-0ef2-42f8-93c2-ff61de672524)

**Reply by SWivid** [reply]
the first picture, the conda installation

**Reply by SyamsQ** [reply]
There is no anything like "Add to PATH". I do it myself "G:\>cd f5-tts"
![image](https://github.com/user-attachments/assets/146d2186-9fb9-4021-b635-c3341517702d)

**Reply by SWivid** [reply]
the Anaconda installation, the pic you give, after the progress bar:
![image](https://github.com/user-attachments/assets/0a367eac-b76d-4b71-998d-0b927b93fb03)

**Reply by SyamsQ** [reply]
Only these.
![image](https://github.com/user-attachments/assets/a1aa1ec3-14bb-439a-b6f6-514d6ca89fc4)
![image](https://github.com/user-attachments/assets/9138d413-65e0-4f43-a392-a373854c82d0)
![image](https://github.com/user-attachments/assets/0ef9466a-b7e5-44e8-aa51-4d88c2cbe785)
![image](https://github.com/user-attachments/assets/fd24317d-6d0c-4955-8c83-add8d16a6755)
![image](https://github.com/user-attachments/assets/eadc0b9b-1b05-4784-89ff-e870698f0bf6)

**Reply by SWivid** [reply]
how about before installation?

![1732736956686](https://github.com/user-attachments/assets/e66cc260-440e-4567-b148-cc06c4d6847c)

i just got a pic like this on the internet (some blogs for anaconda installation

have you click on the first one?

**Reply by SyamsQ** [reply]
No, there is no.

**Reply by SWivid** [reply]
i just download myself the installer of anaconda to make sure

it is exact there the thing instruct you for PATH (by default is not clicked on)

![1732737298158](https://github.com/user-attachments/assets/98fa1dbd-8d56-4aa1-b1a5-b833af2af6d1)

**Reply by SWivid** [reply]
or if you don't want it to PATH,

here in the installer, there is instruction that lead you to use **the Command Prompt and Powershell** that is already added to your windows start menu

**Reply by SyamsQ** [reply]
I have installed it 3 times, there is really no Add Path. 
![Screenshot 2024-11-28 025954](https://github.com/user-attachments/assets/cd2b9ff6-65db-4389-86a5-0359aa8f03a5)
![Screenshot 2024-11-28 030223](https://github.com/user-attachments/assets/40e1d66a-bba2-424d-8501-f6b5de658d03)
![Screenshot 2024-11-28 030210](https://github.com/user-attachments/assets/5fea984e-f8ef-46a4-9a0a-bb96f01e8ade)
![Screenshot 2024-11-28 030142](https://github.com/user-attachments/assets/a995abaa-42d2-4a18-876a-33b835e59935)

**Reply by SyamsQ** [reply]
How is that?
![image](https://github.com/user-attachments/assets/7760b33c-5365-44b2-acb8-6674a9871fe2)
> or if you don't want it to PATH,
> 
> here in the installer, there is instruction that lead you to use **the Command Prompt and Powershell** that is already added to your windows start menu

**Reply by SWivid** [reply]
the anaconda prompt / powershell prompt
both ok

**Reply by SyamsQ** [reply]
What next? What do I write?

**Reply by SWivid** [reply]
just follow the readme instruction, quite clear there

**Reply by SyamsQ** [reply]
has it worked? But my F5-TTS location is on the G drive:  
Your guidance please. Because the C: drive is almost full. That's why I used the G drive:
![image](https://github.com/user-attachments/assets/4a4766af-fb47-4f1e-843e-fe7ee77467b2)
![image](https://github.com/user-attachments/assets/39f79d45-00fa-45fe-8678-5f206628fdd0)

**Reply by SWivid** [reply]
`cd g:`

or if you are installing with option1 (just inference)
it doesn't matter whether you do the cloning or not, `pip install xxx` will have f5-tts installed as a package under you conda pip pkg environment.

if you do option installation, you could simple run `f5-tts_infer-gradio` after installed.

**Reply by SWivid** [reply]
oh, if your c: if full, maybe the anaconda need to be installed in g:

if already installed in g:, would be fine.

but the model ckpts will be cached to c: probably if you are not specifying a local path, about 5gb.
if you want to know how to use local path, check the corresponding readme and issues

generally, i would recommend you to use default setting (autodownload and use ckpts pulled from huggingface)

**Reply by SyamsQ** [reply]
What should I do after installing Torch and Torchaudio?
![image](https://github.com/user-attachments/assets/b0598f02-55fd-4960-9ecf-3dd128e91a96)

**Reply by SyamsQ** [reply]
I tried this command, it failed. `pip install -r requirements.txt`
![image](https://github.com/user-attachments/assets/8748087c-56dd-423b-9e56-e7768afd3959)

I tried this command, it failed too. Please help.
![image](https://github.com/user-attachments/assets/8d08e0b3-66f1-44e1-afea-f9541e887e4f)

**Reply by SWivid** [reply]
There is an up-to-date video given by the same Youtuber you have watched, check it out

**Reply by SyamsQ** [reply]
What are you talking about?

**Reply by SWivid** [reply]
![1732818903844](https://github.com/user-attachments/assets/5b175487-f1ee-4431-a47e-5718a4f28a6f)
well, it is the pic you provide at top

pls follow a updated tutorial, or just follow the readme we provided...

have fun

**Reply by SyamsQ** [reply]
He gave this tutorial [As a pip package (if just for inference)]. While I want to use Local editable (if also do training, finetuning). 

Because I need to add Indonesian and Arabic.

![image](https://github.com/user-attachments/assets/a098ad8b-9ff6-4864-9732-34e8ffe84b33)

But at 12:21, he's already running a `git clone https://github.com/SWivid/F5-TTS.git`

![image](https://github.com/user-attachments/assets/f10bd922-c52c-4e17-8679-73f383eae0c5)

**Reply by SyamsQ** [reply]
up
